compiler/rustc_trait_selection/src/solve/mod.rs

```diff
@@ -317,18 +318,21 @@ impl<'tcx> EvalCtxt<'_, 'tcx> {
         };
 
         // We do no always define opaque types eagerly to allow non-defining uses in the defining scope.
-        if let (DefineOpaqueTypes::No, ty::AliasKind::Opaque) = (define_opaque_types, kind) {
-            if let Some(def_id) = alias.def_id.as_local() {
-                if self
-                    .unify_existing_opaque_tys(
-                        param_env,
-                        OpaqueTypeKey { def_id, args: alias.args },
-                        self.next_ty_infer(),
-                    )
-                    .is_empty()
-                {
-                    return Some(ty);
-                }
+        if let DefineOpaqueTypes::No = define_opaque_types
```

Comment:
This is copied from #118915, since otherwise `Opaque alias-eq ?0` won't reveal the opaque in `Reveal::All` mode.